lib/Scheduler/aco.cpp

```diff
@@ -257,8 +250,9 @@ FUNC_RESULT ACOScheduler::FindSchedule(InstSchedule *schedule_out,
   for (int i = 0; i < pheremone_size; i++)
     pheremone_[i] = 1;
   initialValue_ = 1;
+  std::unique_ptr<InstSchedule> heuristicSched = FindOneSchedule();
```

Comment:
Suggest the use of `auto`. https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable

Reply:
The guidelines say that auto should be used when it improves readability. One of the circumstances where it improves readability that the standards identified was when the type of the object is obvious from context. (e.g. when you are assigning the result of a cast operation to a variable).  He it is not obvious that FindOneSchedule() returns a unique_ptr to an InstSchedule.  I think that use of auto here would make the code less clear and less readable.

Reply:
That's understandable. I'm fine with not using auto here then.